### PR TITLE
Led prop

### PR DESCRIPTION
**Pull request by salminenniklas**
Created development board for ESP32S NodeMCU. It has similar features to the generic ESP32 board in Zephyr although it has onboard LED which we can use for Zephyr samples such as blinky or Zephyr MCUboot port which uses same DTS alias.

### CODE REVIEW

**Comment by dcpleung** [comment]
Please rebase to remove the merge commit.

File: boards/xtensa/esp32s_nodemcu/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ESP32S NodeMCU board configuration
+
+# Copyright (c) 2017 Intel Corporation
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/Kconfig.board:3]: 2022?

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,275 @@
+.. _esp32s_nodemcu:
+
+ESP32S NodeMCU
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:3]: This file was update in latest main. Would you update here as well?

File: boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+identifier: esp32s_nodemcu
+name: ESP32S NodeMCU
+type: mcu
+arch: xtensa
+toolchain:
+  - espressif
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.yaml:6]: Toolchain is now `zephyr`.

File: boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+supported:
+  - gpio
+  - i2c
+  - watchdog
+  - uart
+  - pinmux
+  - nvs
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.yaml:13]: Please, add the following:  
``` c
testing:
  ignore_tags:
    - net
    - bluetooth
```

**Comment by sylvioalves** [comment]
@uyen9vba Only a few comments, thanks for the PR.

File: boards/xtensa/esp32s_nodemcu/Kconfig.board [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# ESP32S NodeMCU board configuration
+
+# Copyright (c) 2022 Intel Corporation
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/Kconfig.board:3]: Changing the date to 2022 but it was not developed by intel? I suggest to rollback to original date, or put the author in the copyright.

File: boards/xtensa/esp32s_nodemcu/Kconfig.defconfig [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# ESP32S NodeMCU board configuration
+
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/Kconfig.defconfig:5]: Ditto

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+ESP32
+#####
+
+Overview
+********
+
+ESP32 is a series of low cost, low power system on a chip microcontrollers
+with integrated Wi-Fi & dual-mode Bluetooth.  The ESP32S NodeMCU employs a
+Tensilica Xtensa LX6 dual-core microprocessor.  ESP32 is created and
+developed by Espressif Systems, a Shanghai-based Chinese company,
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:12]: ESP32S2 is not a LX6 but an LX7 Xtensa, it is also single core

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+and is manufactured by TSMC using their 40nm
+process. [1]_
+
+The features include the following:
+
+- Dual core Xtensa microprocessor (LX6), running at 240MHz
+- 520KB of SRAM
+- 802.11b/g/n/e/i
+- Bluetooth v4.2 BR/EDR and BLE
+- Various peripherals:
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:22]: Please check this information against the esp32s2 reference manual, I'm afraid this is not correct

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: esp32s_nodemcu
+   :goals: debug
+
+Using JTAG
+======================
+
+On the ESP-WROVER-KIT board, the JTAG pins are connected internally to
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:133]: ESP-WROVER-KIT?

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+switch, next to similar headers for SPI and UART.  See
+`ESP-WROVER-32 V3 Getting Started Guide`_ for details.
+
+On the ESP32S NodeMCU board, the JTAG pins are not run to a
+standard connector (e.g. ARM 20-pin) and need to be manually connected
+to the external programmer (e.g. a Flyswatter2):
+
++------------+-----------+
+| ESP32 pin  | JTAG pin  |
++============+===========+
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:148]: I'm afraid this is also not valid for s2 nodemcu.

File: boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.dts [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32.dtsi>
+#include "esp32s_nodemcu-pinctrl.dtsi"
+
+/ {
+	model = "esp32s_nodemcu";
+	compatible = "espressif,esp32";
```
Comment (uLipe) [inline boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.dts:13]: esp32s2?

**Review by uLipe — CHANGES_REQUESTED** [review]
Some points on the documentation please have a check.

File: boards/xtensa/esp32s_nodemcu/doc/index.rst (hunk 7 above)
Comment (sylvioalves) [reply boards/xtensa/esp32s_nodemcu/doc/index.rst:12]: @uLipe This ESP32 based soc, not ESP32S2.

File: boards/xtensa/esp32s_nodemcu/Kconfig.board (hunk 5 above)
Comment (sylvioalves) [reply boards/xtensa/esp32s_nodemcu/Kconfig.board:3]: You can use as an example:  
```c
/*
 * Copyright (c) 2022 Your Name <your@email>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
 ```

File: boards/xtensa/esp32s_nodemcu/doc/index.rst (hunk 7 above)
Comment (uLipe) [reply boards/xtensa/esp32s_nodemcu/doc/index.rst:12]: Got tricked by the s in the board name. My bad

**Review by uLipe — DISMISSED** [review]

**Comment by glaubermaroto** [comment]
Please, apply fixes/suggestions to the motivating commit instead of adding a new one. For example, the fixes in 4e77035bc7f53b798cc8856ab888995675190a50 can be directly added to 3fbcf25597bcec054d1a5fabacb74d3820662f07, 63856146c582b32b629b775d57e635220ce1d430 and f5221bb75072a75ecf68af9e34497d6d65e5465e.

Also, consider squashing related commits into a single one (e.g., 1d6ce640d27fcd317d31317de33e208e7df7f977 and f5221bb75072a75ecf68af9e34497d6d65e5465e). It will avoid cluttering the PR history.

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+======================
+
+On the ESP32S NodeMCU board, the JTAG pins are connected internally to
+a USB serial port on the same device as the console.  These boards
+require no external hardware and are debuggable as-is.  The JTAG
+signals, however, must be jumpered closed to connect the internal
+controller (the default is to leave them disconnected).  The jumper
+headers are on the right side of the board as viewed from the power
+switch, next to similar headers for SPI and UART.  See
+`ESP-WROVER-32 V3 Getting Started Guide`_ for details.
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:140]: I don't think ESP32S NodeMCU board has JTAG-USB converter. So this section needs to either modified or removed. This is valid for esp-wover kit.

File: boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.dts [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+	};
+
+
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/esp32s_nodemcu.dts:29]: Remove extra line.

**Comment by sylvioalves** [comment]
You have to add this board into `pktqueue` sample ignore list:  
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/arch/smp/pktqueue/sample.yaml#L18  

As a suggestion only, I think the PR could be squashed in a single commit with a different title: `boards: esp32: add support to ESP32S NodeMCU`.

File: boards/xtensa/esp32s_nodemcu/doc/index.rst (hunk 12 above)
Comment (uyen9vba) [reply boards/xtensa/esp32s_nodemcu/doc/index.rst]: Removed the JTAG-USB section but left JTAG documentation for using external debugger. I have tested JTAG pinout with Segger JLink Debugger Probe using openOCD software using esp-wroom-32 board configuration.

File: boards/xtensa/esp32s_nodemcu/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,211 @@
+.. _esp32:
```
Comment (sylvioalves) [inline boards/xtensa/esp32s_nodemcu/doc/index.rst:1]: Please, rename to `esp32s_nodemcu`.

**Review by sylvioalves — DISMISSED** [review]

**Comment by sylvioalves** [comment]
@uyen9vba The commit message body requires a description.  
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body

**Pull request closed without merging**